src/components/Option/index.js

```diff
@@ -8,26 +8,19 @@ import StyledOption from './styles/StyledOption'
  * A div which contains a text and which can be selected.
  *
  */
-class Option extends React.PureComponent {
-  state = {
-    selected: this.props.selected,
-  }
+const Option = ({ children, className, id, selected: propsSelected }) => {
+  const [selected, setSelected] = useState(propsSelected)
 
-  toggleSelected = () => {
-    this.setState(prev => ({
-      selected: !prev.selected,
-    }))
-  }
-
-  render() {
-    const { children, className, id } = this.props
-    const { selected } = this.state
-    return (
-      <StyledOption className={className} id={id} onClick={this.toggleSelected} selected={selected}>
-        {children}
-      </StyledOption>
-    )
-  }
+  return (
+    <StyledOption
+      className={className}
+      id={id}
+      onClick={() => setSelected(!selected)}
```

Comment:
Can you pass a reference to a function instead of anonymous one please ?

Reply:
`useCallback` is not needed here unless we have performance issues with options. `useCallback` has a cost at each render : a new function is still created but the ancient reference is kept. So we can use it, but only for optimization with `memo` under it.

Reply:
Good to know, thanks!